### ISSUE TITLE
Tests rely on deprecated SecurityManager

### ISSUE THREAD

**Issue by mizdebsk** [fedora-java/xmvn#132]
When running XMvn tests on Java 17 the following warnings are displayed:

    WARNING: A terminally deprecated method in java.lang.System has been called
    WARNING: System::setSecurityManager has been called by org.fedoraproject.xmvn.it.AbstractIntegrationTest (file:/home/kojan/git/xmvn/xmvn-it/target/test-classes/)
    WARNING: Please consider reporting this to the maintainers of org.fedoraproject.xmvn.it.AbstractIntegrationTest
    WARNING: System::setSecurityManager will be removed in a future release

`setSecurityManager()` method seems to be called from the following 2 files:

* xmvn-it/src/test/java/org/fedoraproject/xmvn/it/AbstractIntegrationTest.java
* xmvn-tools/xmvn-install/src/test/java/org/fedoraproject/xmvn/tools/install/JarUtilsTest.java

**Comment by mizdebsk** [comment]
SecurityManager has been depcecated in [JEP-411](https://openjdk.java.net/jeps/411).
XMvn tests rely on SecurityManager to inhibit `System.exit()` calls. A new feature is planned to allow that in [JDK-8199704](https://bugs.openjdk.java.net/browse/JDK-8199704).

**Comment by mizdebsk** [comment]
> The case in `JarUtilsTest` is there to simulate an error when attempting to write to a file.

We can consider using Byteman to replace this use case of deprecated SecurityManager.

**Comment by mkoncek** [comment]
Using byteman is an option but I was able to come up with a seemingly simpler solution. I checked the debug trace of this test case and with this change, the exception is thrown at line https://github.com/fedora-java/xmvn/blob/5199deebc264f797227d547909d3b87e12fec1ae/xmvn-tools/xmvn-install/src/main/java/org/fedoraproject/xmvn/tools/install/JarUtils.java#L244
i.e. the control flow does not enter the inner try block, I don't know whether we want this or not.
